doc/banner/banner_hld.md

```diff
@@ -0,0 +1,263 @@
+# Banner messages HLD #
+
+## Table of contents
+- [Revision](#revision)
+- [About this manual](#about-this-manual)
+- [Scope](#scope)
+- [1 Introduction](#1-introduction)
+  - [1.1 Feature overview](#11-feature-overview)
+  - [1.2 Requirements](#12-requirements)
+- [2 Design](#2-design)
+  - [2.1 Overview](#21-overview)
+  - [2.2 Flows](#25-flows)
+  - [2.3 CLI](#26-cli)
+      - [2.3.1 Config command group](#231-config-command-group)
+      - [2.3.2 Show command](#232-show-command)
+  - [2.4 YANG model](#24-yang-model)
+- [3 Test plan](#3-test-plan)
+  - [3.1 Unit tests via VS](#31-unit-tests-via-vs)
+
+
+# Revision
+| Rev | Date       | Author              | Description     |
+|:---:|:----------:|:-------------------:|:----------------|
+| 0.1 | 01/02/2023 | Sviatoslav Boichuk  | Initial version |
+
+
+# About this manual
+This document provides a high-level information for the SONiC Banner feature and Banner CLI. It describes high-level behavior, internal definition, design of the commands, syntax, and output definition.
+
+# Scope
+The scope of this document is to cover definition, design and implementation of SONiC Banner feature and Banner CLI.
+The document covers the next CLI:
+1.	Commands to configure banner messages
+2.	Command to display banner settings
+
+<!-- omit in toc -->
+## Abbreviations
+| Term  | Meaning                                   |
+|:------|:------------------------------------------|
+| SONiC | Software for Open Networking in the Cloud |
+| MOTD  | Message of the day                        |
+| DB    | Database                                  |
+| CLI   | Сommand-line Interface                    |
+| YANG  | Yet Another Next Generation               |
+
+<!-- omit in toc -->
+## List of figures
+- [Figure 1: Banner system chart diagram](#figure-1-banner-system-chart-diagram)
+- [Figure 2: Banner init flow](#figure-2-banner-init-flow)
+- [Figure 3: Banner config flow](#figure-3-banner-config-flow)
+- [Figure 4: Banner show configuration](#figure-4-banner-show-configuration)
+
+# 1 Introduction
+
+## 1.1 Feature overview
+The SONiC maintains several messages for communication with users. These messages are associated with login and logout processes.
+
+There are few banners message types:
+|  Type         |  Description                                                                    |
+| :-----------: | :-----------------------------------------------------------------------------: |
+|  login        | Display a banner to the users connected locally or remotely before login prompt |
+|  motd         | Display a banner to the users after login prompt                                |
+|  logout       | Display a logout banner to the users connected locally or remotely              |
+
+
+## 1.2 Requirements
+
+**This feature will support the following functionality:**
+1. Show Banner configuration
+2. Configure Banner
+    1. Feature state
```

Comment:
Is enable / disable of banner feature is supported? If so, unable to locate the respective commands.

Reply:
Updated HLD with feature state enable/disable command (please see section 2.3.1 Config command group)

Reply:
It would be great if below details are added into HLD
1. The behavior when feature state is disabled ==> Is it intended to revert back to default SONiC banner? If not, what is the procedure to revert back to default banner?
2. The behavior of when feature state is enabled ==> Is user configured banner will be applied into system only when state is enabled?
3. The behavior of banner configuration when feature state is disabled
4. The provisions to revert back to default banners independently, looks like currently the feature state disabled option is meant for that (if the assumption is correct). Both banners will be reverted to default. I would suggest to add default option in both command syntax (config login banner default, config motd banner default) and remove the feature enable / disable option. This will provide flexibility to user to manage the banners independently and consistent with other configuration commands to set default values.

Reply:
Updated Flows section in HLD with explanation regarding feature state.
Regarding reverting back to default SONiC banners. AFAIK SONiC OS doesn't have an infra to restore configuration to default values, and whatever configuration you changed - to revert it back to "default" you need to configured "default" value again manually. The same is for this feature.